### PR TITLE
Improve callCloudCode mutation to receive a CloudCodeFunction enum instead of a String

### PR DESCRIPTION
**Pull request by davimacedo**

### CODE REVIEW

File: src/GraphQL/ParseGraphQLSchema.js [hunk 1]
```diff
@@ -53,6 +55,7 @@ class ParseGraphQLSchema {
       databaseController: DatabaseController,
       parseGraphQLController: ParseGraphQLController,
       log: any,
+      appId: String,
```
Comment (omairvaiyani) [inline src/GraphQL/ParseGraphQLSchema.js]: This should be the primitive `string` type

File: src/GraphQL/ParseGraphQLSchema.js [hunk 2]
```diff
@@ -64,20 +67,24 @@ class ParseGraphQLSchema {
     this.log =
       params.log || requiredParameter('You must provide a log instance!');
     this.graphQLCustomTypeDefs = params.graphQLCustomTypeDefs;
+    this.appId = params.appId;
```
Comment (omairvaiyani) [inline src/GraphQL/ParseGraphQLSchema.js]: Is it worth adding the `requiredParameter` check for `appId`?

File: src/GraphQL/ParseGraphQLSchema.js (hunk 1 above)
Comment (davimacedo) [reply src/GraphQL/ParseGraphQLSchema.js]: done

File: src/GraphQL/ParseGraphQLSchema.js (hunk 2 above)
Comment (davimacedo) [reply src/GraphQL/ParseGraphQLSchema.js]: done

**Comment by davimacedo** [comment]
@omairvaiyani I added some tests with weird names and I also improved the code to check the function name and warn in the case it can not be used in the GraphQL Schema.

**Comment by omairvaiyani** [comment]
I actually heavily use `"double-barrelled"` cloud function names in my codebase, for namespacing purposes. I'll have think in a separate PR if there's an elegant solution to it or whether it's either better to use the generic `callCloudCode` or migrate my functions. We may face some pressure on this from other developers too - let's wait and see!

**Comment by davimacedo** [comment]
@omairvaiyani what do you think about removing the `-` automatically?

**Comment by omairvaiyani** [comment]
@davimacedo I have a mild preference for replacing `-` with `_`

**Comment by davimacedo** [comment]
Nice. I've just added a task to the GraphQL project.

**Pull request merged**